SpatialGDK/Source/SpatialGDKTests/Private/RPCContainer/RPCContainerTest.cpp

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "TestDefinitions.h"
+
+#include "ObjectDummy.h"
+#include "ObjectSpy.h"
+
+#include "Utils/RPCContainer.h"
+#include "Schema/RPCPayload.h"
+
+#include <Core.h>
+
+#define RPCCONTAINER_TEST(TestName) \
+	TEST(FRPCContainer, TestName)
+
+using namespace SpatialGDK;
+
+namespace
+{
+	ESchemaComponentType AnySchemaComponentType = ESchemaComponentType::SCHEMA_ClientReliableRPC;
+	ESchemaComponentType AnyOtherSchemaComponentType = ESchemaComponentType::SCHEMA_ClientUnreliableRPC;
+
+	FUnrealObjectRef GenerateObjectRef(UObject* TargetObject)
+	{
+		return FUnrealObjectRef{ Worker_EntityId(TargetObject), 0 };
+	}
+
+	uint32 GeneratePayloadFunctionIndex()
+	{
+		static uint32 FreeIndex = 0;
+		return FreeIndex++;
+	}
+
+	FPendingRPCParamsPtr CreateMockParameters(UObject* TargetObject, ESchemaComponentType Type)
+	{
+		// Use PayloadData as a place to store RPC type
+		RPCPayload Payload(0, GeneratePayloadFunctionIndex(), SpyUtils::TypeToArray(Type));
+		int ReliableRPCIndex = 0;
+
+		FUnrealObjectRef ObjectRef = GenerateObjectRef(TargetObject);
+
+		return MakeUnique<FPendingRPCParams>(ObjectRef, MoveTemp(Payload), ReliableRPCIndex);
+	}
+}
+
+RPCCONTAINER_TEST(GIVEN_a_container_WHEN_nothing_has_been_added_THEN_nothing_is_queued)
+{
+	UObjectDummy* TargetObject = NewObject<UObjectDummy>();
+	FUnrealObjectRef ObjecRef = GenerateObjectRef(TargetObject);
+	FPendingRPCParamsPtr Params = CreateMockParameters(TargetObject, AnySchemaComponentType);
+	FRPCContainer RPCs;
+
+	TestFalse("No queued RPCs of such type", RPCs.ObjectHasRPCsQueuedOfType(ObjecRef.Entity, AnySchemaComponentType));
+
+    return true;
+}
+
+RPCCONTAINER_TEST(GIVEN_a_container_WHEN_one_value_has_been_added_THEN_it_is_queued)
+{
+	UObjectDummy* TargetObject = NewObject<UObjectDummy>();
+	FUnrealObjectRef ObjecRef = GenerateObjectRef(TargetObject);
+	FPendingRPCParamsPtr Params = CreateMockParameters(TargetObject, AnySchemaComponentType);
+	FRPCContainer RPCs;
+
+	RPCs.QueueRPC(MoveTemp(Params), AnySchemaComponentType);
+
+	TestTrue("Queued RPCs of such type", RPCs.ObjectHasRPCsQueuedOfType(ObjecRef.Entity, AnySchemaComponentType));
```

Comment:
I wouldn't know what this string meant without reading the whole function.

Reply:
Also isn't this string printed if it fails?

Reply:
It is printed, besides the test name.

Reply:
Updated.